ofrak_ai/ofrak_ai/sassy_string_modifier.py

```diff
@@ -0,0 +1,187 @@
+import logging
+import openai
+import string
+
+from dataclasses import dataclass, field
+from enum import Enum
+from openai.openai_object import OpenAIObject
+from openai.error import OpenAIError
+from tiktoken import Encoding, encoding_for_model
+from typing import Dict, List, Optional
+
+from ofrak import Resource
+from ofrak.core.strings import AsciiString, StringPatchingConfig, StringPatchingModifier
+from ofrak.component.modifier import Modifier
+from ofrak_ai.chatgpt import ChatGPTConfig
+from ofrak_ai.exponential_backoff import retry_with_exponential_backoff
+
+LOGGER = logging.getLogger(__name__)
+
+
+class StringTypeEnum(Enum):
+    IDENTIFIER = 0
+    SENTENCE = 1
+
+
+@dataclass
+class SassyStringModifierConfig(ChatGPTConfig):
+    """
+    :param min_length: the minimum string length required for targeting strings
+    :param encoding: the tiktoken encoding to use for calculating the number of tokens in a string
+    :param max_retries: the maximum number of attempts to ask ChatGPT to meet the prompt specs
+        before forcefully truncating the response
+    :param prompt_parts: adjustable prompt specifications to give to ChatGPT based on the string
+        type
+    """
+
+    min_length: int = 50
+    encoding: Encoding = encoding_for_model(ChatGPTConfig.model)
+    max_retries: int = 3
+    prompt_parts: Dict[StringTypeEnum, str] = field(default_factory=dict)
+
+
+class SassyStringModifier(Modifier[SassyStringModifierConfig]):
+    """
+    Targets all [AsciiStrings][ofrak.core.strings.AsciiString] over a specified length, requests
+    ChatGPT to sassify them, and patches the sassified strings back into the binary.
+    """
+
+    targets = (AsciiString,)
+
+    async def modify(self, resource: Resource, config: SassyStringModifierConfig):
```

Comment:
All the fields of `SassyStringModifierConfig` have default values, so we can easily provide a default config for this modifier.

```suggestion
    async def modify(self, resource: Resource, config: SassyStringModifierConfig = SassyStringModifierConfig()):
```